src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -117,12 +132,15 @@ export class CampaignActionService implements OnDestroy {
   }
 
   havePreviewParamsChanged(a: Flight, b: Flight) {
-    const check = ['startAt', 'endAt', 'set_inventory_uri', 'zones', 'targets', 'totalGoal', 'dailyMinimum', 'deliveryMode', 'isCompanion'];
-    return check.some(fld => {
-      if (this.hasChanged(b[fld], a[fld])) {
-        return true;
-      }
-    });
+    const check = ['startAt', 'endAt', 'set_inventory_uri', 'targets', 'totalGoal', 'dailyMinimum', 'deliveryMode', 'isCompanion'];
+    return (
```

Comment:
Sometimes the flight zones from the API has pingbacks that the form doesn't know about. The only thing that requires a new preview is add/remove/update a zone id, not urls, not pingbacks. Where we are coercing the InventoryZone into a FlightZone is perhaps also troublesome. I saw where some had labels and others didn't.

Reply:
It would be nice to at some point convert this logic into a pipe-like execution of validation functions that can handle complex logic for comparing changes to values like zones and creatives, which may not need preview updates for every items' value changes.

Reply:
yeah I was also thinking that there should be separate streams for form updates and loading the preview because the logic and model is just getting more and more complex